### ISSUE TITLE
ASP Net Core - Controllers as Services with Attribute Filtering

### ISSUE THREAD

**Issue by oh-yeah** [autofac/Autofac.Extras.AttributeMetadata#13]
## Describe the Bug

How can I add attribute filtering to controllers in in a Net 8 Mvc Project?  Based on the [documentation](https://autofac.readthedocs.io/en/latest/integration/aspnetcore.html#controllers-as-services), controllers need to be configured as service in order to for them to be to be resolved by Autofac.  I'm unable to find any hooks into this registration pipeline for attribute filtering as this registration happens with the service collection.

## Steps to Reproduce

Create a Net8 Mvc Project and register the controllers as services using the following configuration.

```c#
public void ConfigureServices(IServiceCollection services)
{
      services.AddControllersWithViews(o => { /* global filters and auth could be added here */})
              .AddControllersAsServices();
}
```

## Expected Behavior

I should be able to apply attribute filtering, using WithAttributeFiltering(), to these controllers.

## Exception with Stack Trace
```text
N/A
```

## Dependency Versions

Autofac: 7.1.0
Autofac.Extensions.DependencyInjection: 8.0.0.0
Autofac.Extras.AttributeMetadata: 6.0.0.0

**Comment by tillig** [comment]
You probably noticed that, on the issue template, "usage questions" directs you [to Stack Overflow](https://stackoverflow.com/questions/tagged/autofac). There are a lot of reasons for that, including:

- Only one or two people monitor issues in an as-free-time-allows basis. The whole Autofac and ASP.NET Core community monitors Stack Overflow - including all the folks who see issues here.
- We can't really provide every combinatoric explanation for how feature X integrates with framework Y to accomplish outcome Z. I wish we had that kind of time. We rely on the community to provide pull requests and updated examples or documentation to help everyone out.

I do not have an answer to this. I know you can _re-register_ your controllers in the `ConfigureContainer` method. This will register the controllers _after_ the version registered by .NET Core and may offer you an avenue. However, I can't say I've tried this. I'd bet someone else out in the community has.

Hence... I'm going to close this, and I encourage you to follow the issue template and head over to Stack Overflow for this instead.

**Comment by oh-yeah** [comment]
After some research we've decided to refactor our code and put the dependencies that require attribute filtering into a service.  Honestly, that's probably the right solution for our problem here.  That being said, I do see this as a missing feature as all topics I've found on modifying an existing registration resulted in overly complex code that ultimately just replaces the instance that's being resolved.  Overriding the existing registrations, or re-registering, doesn't seem like the correct approach either as we'd lose some of the configurations that only are available withing the context of registering controllers with the service collection (ie, exception filters, auth, etc).
```c#
services.AddControllers(o =>
{
     o.Filters.Add<MyExceptionFilter>();
     o.Filters.Add(new AuthorizeFilter(new AuthorizationPolicyBuilder().RequireAuthenticatedUser().Build()));
 });
```
What's really needed is the ability to alter registrations when items from the ServiceCollection are populated into the ContainerBuilder.  This kind of configuration was extremely easy to do in .NET Framework projects as we didn't to deal with the complexity of different containers nor did we have the same level of configuration on controllers themselves.
```c#
builder.RegisterApiControllers(Assembly.GetExecutingAssembly()).WithAttributeFiltering();
```
Maybe this isn't the correct project for this feature request?  This feels more like something that should be requested on the Autofac.Extensions.DependencyInjection project.  I can spend some time writing something up if that's the right project, but if it feels like a niche scenario that wouldn't get traction then there's no need to waste anyone's time.
